core/src/main/scala/org/bitcoins/core/crypto/SchnorrDigitalSignature.scala

```diff
@@ -0,0 +1,71 @@
+package org.bitcoins.core.crypto
+
+import org.bitcoins.core.protocol.NetworkElement
+import org.bitcoins.core.util.{BitcoinSLogger, BitcoinSUtil, Factory}
+import scodec.bits.ByteVector
+
+sealed abstract class SchnorrDigitalSignature extends NetworkElement {
+  require(rx.length == 32, s"R must be 32 bytes, got $rx")
+  require(s.length == 32, s"s must be 32 bytes, got $s")
+  require(bytes.length == 64, s"This must be 64 bytes, got $bytes")
+
+  /** Represents the x coordinate of the R value used in the signature */
+  def rx: ByteVector
+
+  /** The s = k + e*x value of the signature */
+  def s: ByteVector
+
+  override def bytes: ByteVector = rx ++ s
+
+  override def toString: String = s"SchnorrDigitalSignature($rx, $s)"
+}
+
+final case object DummySchnorrDigitalSignature extends SchnorrDigitalSignature {
+  override val rx: ByteVector = ByteVector.fill(32)(0.toByte)
+  override val s: ByteVector = rx
+}
+
+object SchnorrDigitalSignature extends Factory[SchnorrDigitalSignature] {
+  private case class SchnorrDigitalSignatureImpl(rx: ByteVector, s: ByteVector)
+      extends SchnorrDigitalSignature
+
+  override def fromBytes(bytes: ByteVector): SchnorrDigitalSignature = {
+    if (bytes.length != 64) {
```

Comment:
`require(bytes.length == 64,  s"SchnorrDigitalSignature must be 64 bytes, got $bytes")`